src/data-restriction/DataRestrictionModal.tsx

```diff
@@ -0,0 +1,214 @@
+        more about the study and how to request access to the data.
+      </p>
+    </div>
+  ) : (
+    <div>
+      <h2>The {safeHtml(studyName)} study has data access restrictions.</h2>
+      <hr />
+    </div>
+  );
+}
+
+function PolicyNotice(props: Props) {
+  const { study, user, permissions, action } = props;
+  const message = getRestrictionMessage({ action, permissions, study, user });
+  const policyUrl = getPolicyUrl(study);
+  return isPrereleaseStudy(
+    getStudyAccess(study),
+    getStudyId(study),
+    user,
+    permissions
+  ) ? null : !policyUrl ? null : getRequestNeedsApproval(study) == '0' ? (
+    <p>
+      {message} Data access will be granted immediately upon request submission.
+      <br />
+      <br />
+      Please read the{' '}
+      <a href={policyUrl} target="_blank" rel="noreferrer">
+        {' '}
+        Data Access and Use Policy.
+      </a>
+    </p>
+  ) : (
+    <p>
+      {message} The data from this study requires approval to download and use
+      in research projects.
+      <br />
+      <br />
+      Please read the{' '}
+      <a href={policyUrl} target="_blank" rel="noreferrer">
+        {' '}
+        Data Access and Use Policy.
+      </a>
+    </p>
+  );
+}
+
+function Buttons(props: Props) {
+  const { action, study, user, permissions, showLoginForm, onClose } = props;
+  const history = useHistory();
+  const strict = isActionStrict(action);
+  const approvalRequired = actionRequiresApproval({
+    action,
+    permissions,
+    study,
+    user,
+  });
+  return isPrereleaseStudy(
+    getStudyAccess(study),
+    getStudyId(study),
+    user,
+    permissions
+  ) ? (
+    <div className="DataRestrictionModal-Buttons">
+      {!strict ? (
+        <button className="btn" onClick={onClose}>
+          Dismiss
+          <Icon fa="times right-side" />
+        </button>
+      ) : window.history.length > 1 ? (
+        <button
+          className="btn"
+          title="Go Back"
+          type="button"
+          onClick={() => window.history.go(-1)}
+        >
+          Go Back
+          <Icon fa="long-arrow-left right-side" />
+        </button>
+      ) : (
+        <Link className="btn" to="/" title="Go Home">
+          Go Home
+          <Icon fa="home right-side" />
+        </Link>
+      )}
+    </div>
+  ) : (
+    <div className="DataRestrictionModal-Buttons">
+      {!user.isGuest ? null : (
+        <button
+          onClick={() => showLoginForm(window.location.href)}
+          className="btn"
+        >
+          Log In
+          <Icon fa="sign-in right-side" />
+        </button>
+      )}
+      {!approvalRequired ? null : (
+        <button
+          onClick={() => {
+            const loggedInRoute = `/request-access/${getStudyId(
+              study
+            )}?redirectUrl=${encodeURIComponent(window.location.href)}`;
+
+            if (user.isGuest) {
+              showLoginForm(history.createHref(parsePath(loggedInRoute)));
```

Comment:
Because `loggedInRoute` is relative, this "gotcha" I [observed in another PR](https://github.com/VEuPathDB/EbrcWebsiteCommon/pull/96#discussion_r809667036) might be relevant.